src/System.Memory/tests/ReadOnlySpan/GetPinnableReference.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class ReadOnlySpanTests
+    {
+        [Fact]
+        public static void GetPinnableReferenceArray()
+        {
+            int[] a = { 91, 92, 93, 94, 95 };
+            ReadOnlySpan<int> span = new ReadOnlySpan<int>(a, 1, 3);
+            ref int pinnableReference = ref Unsafe.AsRef(in span.GetPinnableReference());
+            Assert.True(Unsafe.AreSame(ref a[1], ref pinnableReference));
+        }
+
+        // Enable this test once we use C# 7.3
+        //[Fact]
+        //public static unsafe void UsingSpanInFixed()
+        //{
+        //    byte[] a = { 91, 92, 93, 94, 95 };
+        //    ReadOnlySpan<byte> span = a;
+        //    fixed (byte* ptr = span)
+        //    {
+        //        for (int i = 0; i < span.Length; i++)
+        //        {
+        //            Assert.Equal(a[i], ptr[i]);
+        //        }
+        //    }
+        //}
+
+        // Enable this test once we use C# 7.3
+        //[Fact]
+        //public static unsafe void UsingEmptySpanInFixed()
+        //{
+        //    ReadOnlySpan<int> span = ReadOnlySpan<int>.Empty;
+        //    fixed (int* ptr = span)
+        //    {
+        //        Assert.True(ptr == null);
+        //    }
+
+        //    ReadOnlySpan<int> spanFromEmptyArray = Array.Empty<int>();
+        //    fixed (int* ptr = spanFromEmptyArray)
+        //    {
+        //        Assert.True(ptr == null);
+        //    }
+        //}
+
+        [Fact]
+        public static void GetPinnableReferenceArrayPastEnd()
+        {
+            // The only real difference between GetPinnableReference() and "ref span[0]" is that
+            // GetPinnableReference() of a zero-length won't throw an IndexOutOfRange but instead return a null ref.
+
+            int[] a = { 91, 92, 93, 94, 95 };
+            ReadOnlySpan<int> span = new ReadOnlySpan<int>(a, a.Length, 0);
+            ref int pinnableReference = ref Unsafe.AsRef(in span.GetPinnableReference());
+            ref int expected = ref Unsafe.Add<int>(ref a[a.Length - 1], 1);
+            Assert.True(Unsafe.AreSame(ref expected, ref pinnableReference));
+        }
+
+        [Fact]
+        public static void GetPinnableReferencePointer()
+        {
+            unsafe
```

Comment:
Why in some cases you make the method unsafe and in others you have an unsafe bloc?

Reply:
No reason other than copy/pasted from the MemoryMarshal.GetReference tests: https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/MemoryMarshal/GetReference.cs#L83